uaclient/apt_news.py

```diff
@@ -90,6 +103,54 @@ def do_selectors_apply(
         if selectors.pro != _is_attached(cfg).is_attached:
             return False
 
+    if selectors.architectures is not None:
+        if system.get_dpkg_arch() not in selectors.architectures:
+            return False
+
+    if selectors.packages is not None:
+        installed_packages = get_installed_packages()
+        installed_packages_names = [
+            package.name for package in installed_packages
+        ]
+        is_package_matched = False
+
+        for package in selectors.packages:
+            try:
+                package_name, version_operator, package_version = package
+            except ValueError:
+                LOG.warning("Invalid package selector: %r", package)
+                return False
+            if not is_installed(package_name):
+                LOG.info("Package %s is not installed", package_name)
+                return False
+            installed_package = installed_packages[
+                installed_packages_names.index(package_name)
+            ]
+            version_comparison = version_compare(
+                installed_package.version, package_version
+            )
+            if any(
+                [
+                    (
+                        version_comparison == 0
+                        and version_operator in ["==", "<=", ">="]
+                    ),
+                    (
+                        version_comparison < 0
+                        and version_operator in ["<", "<="]
+                    ),
+                    (
+                        version_comparison > 0
+                        and version_operator in [">", ">="]
+                    ),
+                ]
+            ):
+                is_package_matched = True
+                break
```

Comment:
I think we can update the logic to a `not any` and just return `False` here.
Then we don't need the `is_package_matched` variable anymore